src/AutoLockScrolling.jsx

```diff
@@ -0,0 +1,65 @@
+import React from 'react';
+
+let originalBodyOverflow = null;
+let lockingCounter = 0;
+
+export default class AutoLockScrolling extends React.Component {
+
+  static propTypes = {
+    lock: React.PropTypes.bool.isRequired,
+  };
+
+  constructor(props) {
+    super(props);
+  }
+
+  componentDidMount() {
+    if (this.props.lock) {
+      this.applyAutoLockScrolling(this.props.lock);
+    }
+  }
+
+  componentWillReceiveProps(nextProps) {
+    if (this.props.lock !== nextProps.lock) {
+      this.applyAutoLockScrolling(nextProps.lock);
+    }
+  }
+
+  componentWillUnmount() {
+    if (this.props.lock === true) {
+      this.allowScrolling();
+    }
+  }
+
+  applyAutoLockScrolling(lock) {
+    if (lock) {
+      this.preventScrolling();
+    } else {
+      this.allowScrolling();
+    }
+  }
+
+  preventScrolling() {
+    lockingCounter = lockingCounter + 1;
+    //only lock the first time the component is mounted.
+    if (lockingCounter === 1) {
+      const body = document.getElementsByTagName('body')[0];
+      originalBodyOverflow = body.style.overflow;
+      body.style.overflow = 'hidden';
+    }
+  }
+
+  allowScrolling() {
+    lockingCounter = lockingCounter - 1;
```

Comment:
I have a bad feeling that if we switch the lock property on/off while the component is mounted it might go wrong. not sure.

I have a plane to take, so I'll have to check it again carefully later.